lib/preview/httpsRedirectingServer.js

```diff
@@ -0,0 +1,85 @@
+const http = require('http');
+const https = require('https');
+const net = require('net');
+const { unlinkSync } = require('fs');
+const { silenceConsoleWhile } = require('./previewUtils');
+
+const createCert = async (domainsToProxy) => {
+  const additionalMkcertHosts = domainsToProxy
+    .map(proxyDomain => [
+      `${proxyDomain.domain}.localhost`,
+      `${proxyDomain.domain}.hslocal.net`
+    ])
+    .flat();
+  const hosts = ['localhost', 'hslocal.net', ...additionalMkcertHosts];
+  const { createCertificate } = await import('mkcert-cli');
```

Comment:
What's the reasoning for the dynamic import?

Reply:
https://github.com/HubSpot/cli-lib/pull/21#discussion_r1476752068

See ^